nova/cmd/manage.py

```diff
@@ -831,8 +832,19 @@ def purge_request_specs(self, dry_run=False, older_than=90, max_number=0,
         admin_context = context.get_admin_context()
         # There's no need to join the default security_groups and
         # instance_info_caches tables, hence explicit empty 'columns_to_join'.
-        extant_instance_uuids = [i.get('uuid') for i in db.instance_get_all(
-            admin_context, columns_to_join=[]) if i.get('uuid') is not None]
+        extant_instance_uuids = set(i.get('uuid') for i in db.instance_get_all(
+            admin_context, columns_to_join=[]) if i.get('uuid') is not None)
+        # Fetch instances from other cells than the default cell1 and cell0
+        extra_cells = [cell for cell
+                       in cell_mapping_obj.CellMappingList.get_all(
+                            admin_context)
+                       if cell.name not in ["cell0", "cell1"]]
+        for cell in extra_cells:
```

Comment:
Why filter some cells out (hardcoded) here instead of looping over all cells to begin with?

Reply:
True. Fixed.